File: Arcade/TheCore/Javascript/squareDigitsSequence.js
function squareDigitsSequence(a0) {
    const seq = [a0]
    
    while (true) {
        const sum = [...seq[0]+""]
            .map(i => Math.pow(+i, 2))
            .reduce((total, curr) => total + curr);
            
        if (seq.includes(sum)) break;
        
        seq.unshift(sum);
    }
    
    return seq.length + 1;
}

File: Arcade/TheCore/Kotlin/squareDigitsSequence.kt
fun squareDigitsSequence(a0: Int): Int {
    val seq: MutableList<Int> = mutableListOf(a0);
    
    while (true) {
        val sum: Int = (""+seq[0])
            .map { Math.pow((""+it).toInt().toDouble(), 2.0) }
            .sumOf { it.toInt() }
            
        if (sum in seq) break;        
        
        seq.add(0, sum);
    }
    
    return seq.size + 1;
}